Sources/Core/AWSClientRuntime/Retry/AWSRetryErrorInfoProvider.swift

```diff
@@ -0,0 +1,109 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import struct Foundation.TimeInterval
+import enum AwsCommonRuntimeKit.CommonRunTimeError
+import protocol ClientRuntime.RetryErrorInfoProvider
+import enum ClientRuntime.DefaultRetryErrorInfoProvider
+import struct ClientRuntime.RetryErrorInfo
+import protocol ClientRuntime.ServiceError
+import protocol ClientRuntime.HTTPError
+import protocol ClientRuntime.ModeledError
+
+public enum AWSRetryErrorInfoProvider: RetryErrorInfoProvider {
```

Comment:
Tests for certain AWS-specific error codes and HTTP codes, then handles errors as appropriate.

If this type doesn't match an error, it is handed off to the default error info provider for further matching.